client/src/components/EditProfile/EditProfileTab/EditProfileTab.tsx

```diff
@@ -34,57 +33,64 @@ const months = [
-  const [lastName, setLastName] = useState<UserProfile['lastName'] | null>();
-  const [gender, setGender] = useState<UserProfile['gender'] | null>('gender');
-  const [birthdateMonth, setBirthdateMonth] = useState<UserProfile['birthdateMonth'] | null>('month');
-  const [birthdateDay, setBirthdateDay] = useState<UserProfile['birthdateDay'] | null>('day');
-  const [birthdateYear, setBirthdateYear] = useState<UserProfile['birthdateYear'] | null>('year');
-  const [email, setEmail] = useState<UserProfile['email'] | null>();
-  const [phone, setPhone] = useState<UserProfile['phone'] | null>();
-  const [address, setAddress] = useState<UserProfile['address'] | null>();
-  const [description, setDescription] = useState<UserProfile['description'] | null>();
+  const [accountType, setAccountType] = useState<string>('partner');
   const [showPhoneInput, setShowPhoneInput] = useState(false);
+  const [profile, setProfile] = useState<ProfileInterface>({
+    firstName: '',
+    lastName: '',
+    description: '',
+    address: '',
+    phoneNumber: '',
+    dateOfBirth: new Date('December 17, 1995 03:24:00'),
+    available: false,
+    availability: '',
+    gender: '',
+    email: '',
+  });
+  const userId = '619c1eb37a1e963a5b179c4b';
+
+  useEffect(() => {
+    fetchProfile(userId).then((data) => setProfile(data.success.profile)); //get profileValues and set to profile state
+  }, []);
 
   const formik = useFormik({
     initialValues: {
-      available: false,
-      availability: 'availability',
-      firstName: '',
-      lastName: '',
-      gender: 'gender',
-      birthdateMonth: 'month',
-      birthdateDay: 'day',
-      birthdateYear: 'year',
-      email: '',
-      phone: '',
-      address: '',
-      description: '',
+      available: profile.available,
+      availability: profile.availability || 'availability',
+      firstName: profile.firstName,
+      lastName: profile.lastName,
+      gender: profile.gender || 'gender',
+      birthdateMonth: profile.dateOfBirth ? months[new Date(profile.dateOfBirth).getMonth()] : 'month',
+      birthdateDay: profile.dateOfBirth ? new Date(profile.dateOfBirth).getDate() : 'day',
+      birthdateYear: profile.dateOfBirth ? new Date(profile.dateOfBirth).getFullYear() : 'year',
+      email: profile.email,
+      phoneNumber: profile.phoneNumber,
+      address: profile.address,
+      description: profile.description,
     },
     onSubmit: (values) => {
-      if (accountType === 'partner') {
-        setAvailable(values.available);
-        setAvailability(values.availability);
-      }
-      setFirstName(values.firstName);
-      setLastName(values.lastName);
-      setGender(values.gender);
-      setBirthdateMonth(values.birthdateMonth);
-      setBirthdateDay(values.birthdateDay);
-      setBirthdateYear(values.birthdateYear);
-      setEmail(values.email);
-      setPhone(values.phone);
-      setAddress(values.address);
-      setDescription(values.description);
+      edit(
```

Comment:
Optional: You can move this function outside of the component for increase maintainability and readability 👍🏽